tests/test_api.py

```diff
@@ -10,6 +14,28 @@
                        WRONG_MECH, WRONG_PIN)
 
 
+def test_sc_export_pub_key_valid_key_id_should_return_pem(pkcs11):
+  with open(str(Path(__file__).parent / 'keys/public_key.pem'), 'rb') as pem:
+    pub_key_pem = sc_export_pub_key_pem(VALID_KEY_ID, VALID_PIN, pkcs11=pkcs11)
+    assert pub_key_pem == pem.read()
+
+
+def test_sc_export_pub_key_wrong_key_id_should_raise_error(pkcs11):
+  with pytest.raises(SmartCardFindKeyObjectError):
+    sc_export_pub_key_pem(WRONG_KEY_ID, VALID_PIN, pkcs11=pkcs11)
+
+
+def test_sc_export_x509_valid_key_id_should_return_cert(pkcs11):
+  with open(str(Path(__file__).parent / 'keys/x509_cert.pem'), 'rb') as pem:
+    x509_cert = sc_export_x509_pem(VALID_KEY_ID, VALID_PIN, pkcs11=pkcs11)
+    assert x509_cert == pem.read()
+
+
+def test_sc_export_x509_valid_key_id_should_should_raise_error(pkcs11):
+  with pytest.raises(SmartCardFindKeyObjectError):
+    sc_export_x509_pem(WRONG_KEY_ID, VALID_PIN, pkcs11=pkcs11)
```

Comment:
Have you added a test for valid key id wrong pin?

Reply:
Not needed, test [coverage](https://travis-ci.com/openlawlibrary/oll-sc/builds/113457187#L451) for the API is 100%.

Reply:
Ah, I see, session will raise an error. That's fine then.